httpmw/httpmw_test.go

```diff
@@ -0,0 +1,79 @@
+package httpmw
+
+import (
+	"bytes"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"time"
+
+	"github.com/gorilla/mux"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Httpmw", func() {
+	var (
+		router *mux.Router
+	)
+
+	BeforeEach(func() {
+		router = mux.NewRouter()
+	})
+
+	Describe("PanicMiddleware", func() {
+		It("should handle panic and return Internal Server Error", func() {
+			logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
+			router.Use(NewPanicHandling(logger, customPanicHandler))
+			router.HandleFunc("/", func(w http.ResponseWriter, e *http.Request) {
+				panic("oops")
+			})
+
+			rw := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/", nil)
+
+			router.ServeHTTP(rw, req)
```

Comment:
This is not the typical way of test http middleware.

Examples:

 ```
    // Create a new router and apply the middleware.
    r := mux.NewRouter()
    r.Use(LoggingMiddleware)
    r.Handle("/", testHandler)

    // Create a test HTTP server.
    ts := httptest.NewServer(r)
    defer ts.Close()

    // Make a request to the test server.
    resp, err := http.Get(ts.URL)
    if err != nil {
        t.Fatalf("Failed to make GET request: %v", err)
    }
    defer resp.Body.Close()

    // Check the response status code.
    if resp.StatusCode != http.StatusOK {
        t.Errorf("Expected status code %d, got %d", http.StatusOK, resp.StatusCode)
    }

    // Check the response body.
    body := new(strings.Builder)
    _, err = body.ReadFrom(resp.Body)
    if err != nil {
        t.Fatalf("Failed to read response body: %v", err)
    }
    if body.String() != "Test passed" {
        t.Errorf("Expected response body 'Test passed', got '%s'", body.String())
    }
```

Reply:
Followed the standard set by Fleet team for this test: https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/fleet/pkg/api/middleware/panic_handling_mw_test.go&version=GBmaster&_a=contents

Can adjust test if needed.